### PR TITLE
chore(init): init from template

### PR DESCRIPTION
**Pull request by JaeAeich**
Use [cookiecutter](https:github.com/elixir-cloud-aai/cookiecutter-python.git) to initialize the project.

## Summary by Sourcery

Initialize the `tus-storagehandler` project using the cookiecutter template. This includes setting up the project structure, adding CI/CD workflows, documentation, initial tests, and various configuration files for code quality and security checks.

CI:
- Add GitHub Actions workflows for code quality, PR validation, code testing, documentation checks, and vulnerability tests.
- Add custom GitHub Action for setting up Python and Poetry environments.

Deployment:
- Add GitHub Actions workflow for releasing package to PyPI.

Documentation:
- Revamp README.md with new project structure, badges, and detailed sections for usage, installation, development, contributing, code of conduct, versioning, license, and contact information.
- Add Sphinx configuration and Makefile for generating project documentation.
- Add .readthedocs.yaml for Read the Docs configuration.

Tests:
- Add initial unit and integration tests for the project structure.
- Add pre-commit configuration for various code quality checks.

Chores:
- Initialize project structure using cookiecutter template.
- Add .markdownlint.yaml and .yamllint.yaml for linting markdown and YAML files.
- Add .safety-policy.yml for configuring safety checks.
- Add PULL_REQUEST_TEMPLATE.md and ISSUE_TEMPLATE for standardizing contributions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,0 +1,178 @@
+[![Bandit](https://img.shields.io/badge/security-bandit-yellow.svg)](https://bandit.readthedocs.io/en/latest/)
+[![codecov](https://codecov.io/gh/elixir-cloud-aai/tus-storagehandler/branch/main/graph/badge.svg)](https://codecov.io/gh/elixir-cloud-aai/tus-storagehandler)
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)][code-of-conduct]
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply README.md:3]: it is already there!!

File: docs/source/pages/tus_storagehandler.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+tus\_storagehandler package
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply docs/source/pages/tus_storagehandler.rst:1]: I would not touch it as it is auto-generated. Should be fixed later if it cause any issue in future.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,178 @@
...
+- [License](#license)
+- [Contact](#contact)
+
+## Basic Usage
+
+## Installation
+
+## Development
+
+For ease of use, certain scripts have been abbreviated in `Makefile`, make sure
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply README.md:32]: These changes have been tested in cookiecutter and if changes need to be made, they will be made in the template.

**Comment by JaeAeich** [comment]
@uniqueg Please add `codecov` tokens for this repo!! Otherwise the CI will fail.

**Review by uniqueg — COMMENTED** [review]
Let's discuss at the meeting with @psankhe28

File: README.md [hunk 4]
```diff
@@ -110,10 +104,6 @@ make tc
 make pc
 ```
 
-- Update the cookiecutter template
-
-```sh
-make u
```
Comment (JaeAeich) [inline README.md]: why remove this?

File: README.md [hunk 5]
```diff
@@ -175,4 +165,4 @@ To get in touch with us, please use one of the following routes:
 [license]: LICENSE
 [logo-elixir]: images/logo-elixir.svg
 [logo-elixir-cloud-aai]: images/logo-elixir-cloud-aai.svg
-[semver]: https://semver.org/
```
Comment (JaeAeich) [inline README.md]: keep it in, ig it must be referenced somewhere.

File: README.md [hunk 6]
```diff
@@ -14,19 +14,13 @@ File handler utilizing TUS and MinIO with DRS-Filer integration.
...
 - [Contributing](#contributing)
 - [Code of Conduct](#code-of-conduct)
 - [Versioning](#versioning)
 - [License](#license)
 - [Contact](#contact)
 
-## Basic Usage
-
-## Installation
-
```
Comment (JaeAeich) [inline README.md]: same leave it blank, this might be fillerd later.

File: README.md (hunk 4 above)
Comment (psankhe28) [reply README.md]: Updating the template might not be required for this project once itis created, right?

**Comment by JaeAeich** [comment]
@psankhe28 I meant add it from teh readme from main, I don't think you need to remove anything from the readme.

I meant adding there instructions
```
TUS Storage Handler
Synopsis
This Flask application provides endpoints for uploading, downloading, and listing files in a MinIO bucket, with TUS protocol support for uploads, and CORS enabled for cross-origin requests.

Installation
Prerequisites
This flask application requires a running instance of [minio](https://min.io/download)

Run the minio instance by executing the following command in the location where minio is installed

minio server /data --console-address ":9001"

Download the required dependencies

Navigate to the folder TusStorageHandler
Create a virtual environment and activate it (optional)
pip install -r requirements.txt
Running the application
flask run The application will be running on http://127.0.0.1:5000 by default.
```

If they are not needed or have changed, ig better revert it, unless ofc you have a reason.

**Review by uniqueg — DISMISSED** [review]

**Comment by JaeAeich** [comment]
Hey @psankhe28 spell check failed, spelling for synopsis is wrong 😅

@uniqueg have up not added the token or there is something wrong with ci?

**Comment by uniqueg** [comment]
> Hey @psankhe28 spell check failed, spelling for synopsis is wrong 😅
> 
> @uniqueg have up not added the token or there is something wrong with ci?

Indeed, `CODECOV_TOKEN` was not set - it is now, and tests pass (`PYPI_TOKEN` is set organization-wide).

BUT, @JaeAeich: I don't understand the `AUTO_UPDATE_GITHUB_TOKEN`. The linked documentation does not make sense, as it is specific to `GITHUB_TOKEN`, which is _auto-generated_ (and destroyed) by GitHub for each job (see, e.g., [here](https://dev.to/github/the-githubtoken-in-github-actions-how-it-works-change-permissions-customizations-3cgp)): 

image

It seems to me that either [the Action that uses `AUTO_UPDATE_GITHUB_TOKEN`](https://github.com/elixir-cloud-aai/cookiecutter-python/blob/74890427a2926b6943577df5f23001a65325f591/%7B%7B%20cookiecutter.project_name%20%7D%7D/.github/workflows/update.yaml#L48) should use `GITHUB_TOKEN` instead - the settings are configured correctly as per the linked instructions:

image

Or the instructions should instead describe how (and with what exact permissions) `AUTO_UPDATE_GITHUB_TOKEN` should be created.

Do you want to test and change that before or after merging this PR?

**Review by uniqueg — APPROVED** [review]
I'm approving, despite https://github.com/elixir-cloud-aai/tus-storagehandler/pull/10#issuecomment-2270618221, as we can always fix the `update.yaml` later, if necessary.

**Pull request merged**